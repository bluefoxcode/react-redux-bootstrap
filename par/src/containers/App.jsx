import React, { Component } from 'react';
import { Provider } from 'react-redux';
import store, { history } from '../store';
import NotFound from '../components/NotFound';
import { ConnectedRouter } from 'react-router-redux';
import { Route, Switch } from 'react-router';
import { AppContainer } from 'react-hot-loader';

import Main from './Main';

import '../assets/styles/App.scss';

class App extends Component {
	render() {
		return (
			<AppContainer>
				<Provider store={store}>
					<ConnectedRouter history={history}>
						<Switch>
							<Route exact path="/" component={Main} />
							       <Route component={NotFound} />
						</Switch>
					</ConnectedRouter>
				</Provider>
			</AppContainer>
		);
	}
}

export default App;
