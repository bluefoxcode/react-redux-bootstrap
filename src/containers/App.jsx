import React, { Component } from 'react';
import { Button, Container, Header } from 'semantic-ui-react'

import logo from '../assets/logo.svg';
import '../assets/styles/App.scss';

import NavContainer from './Nav';

class App extends Component {
  render() {
    return (
      <div className="App">
        <NavContainer />
        <Container>
          <Header as='h1'>Hello world!</Header>
          <Button
            content='Discover docs'
            href='http://react.semantic-ui.com'
            icon='github'
            labelPosition='left'
          />
        </Container>
        <div className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <h2>Welcoe to React</h2>
        </div>
        <p className="App-intro">
          To get strted, edit <code>src/App.js</code> and save to reload.
        </p>
      </div>
    );
  }
}

export default App;
