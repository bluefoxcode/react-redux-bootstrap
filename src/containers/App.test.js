import React from 'react';
import { render } from 'react-dom';
import { shallow, mount } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import App from './App';

describe('App Component', () => {
	it('matches snapshot', () => {
		const wrapper = shallow(<App />);
		const tree = shallowToJson(wrapper);
		expect(tree).toMatchSnapshot();
	});

	it('renders without crashing', () => {
		const div = document.createElement('div');
		render(<App />, div);
	});

	describe('method someMethod', () => {
		it('returns appropriate string', () => {
			const wrapper = shallow(<App />);
			expect(wrapper.instance().someMethod()).toEqual('Some strings');
		});
	});
});
