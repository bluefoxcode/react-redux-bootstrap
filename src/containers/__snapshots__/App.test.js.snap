// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component matches snapshot 1`] = `
<AppContainer>
  <Provider
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  >
    <ConnectedRouter
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <Switch>
        <Route
          component={[Function]}
          exact={true}
          path="/"
        />
        <Route
          component={[Function]}
        />
      </Switch>
    </ConnectedRouter>
  </Provider>
</AppContainer>
`;
